.NET/Microsoft.Recognizers.Text.DateTime/Parsers/CJK/BaseCJKDatePeriodParser.cs

```diff
@@ -0,0 +1,1622 @@
+                            ret.FutureValue = ret.PastValue = new Tuple<DateObject, DateObject>(beginDate, endDate);
+                            ret.Success = true;
+                            return ret;
+                        }
+
+                        prefixMatch = this.config.FutureRegex.MatchExact(beforeStr, trim: true);
+
+                        if (prefixMatch.Success)
+                        {
+                            DateObject beginDate, endDate;
+                            switch (unitStr)
+                            {
+                                case Constants.TimexDay:
+                                    beginDate = referenceDate;
+                                    endDate = referenceDate.AddDays(number);
+                                    break;
+                                case Constants.TimexWeek:
+                                    beginDate = referenceDate;
+                                    endDate = referenceDate.AddDays(7 * number);
+                                    break;
+                                case Constants.TimexMonthFull:
+                                    beginDate = referenceDate;
+                                    endDate = referenceDate.AddMonths(number);
+                                    break;
+                                case Constants.TimexYear:
+                                    beginDate = referenceDate;
+                                    endDate = referenceDate.AddYears(number);
+                                    break;
+                                default:
+                                    return ret;
+                            }
+
+                            ret.Timex =
+                                $"({DateTimeFormatUtil.LuisDate(beginDate.AddDays(1))},{DateTimeFormatUtil.LuisDate(endDate.AddDays(1))},P{number}{unitStr[0]})";
+                            ret.FutureValue =
+                                ret.PastValue =
+                                    new Tuple<DateObject, DateObject>(beginDate.AddDays(1), endDate.AddDays(1));
+                            ret.Success = true;
+                            return ret;
+                        }
+                    }
+                }
+            }
+
+            return ret;
+        }
+
+        // case like "三月的第一周"
+        private DateTimeResolutionResult ParseWeekOfMonth(string text, DateObject referenceDate)
+        {
+            var ret = new DateTimeResolutionResult();
+            var trimmedText = text.Trim();
+
+            var match = this.config.WeekOfMonthRegex.Match(text);
+            if (!match.Success)
+            {
+                return ret;
+            }
+
+            var cardinalStr = match.Groups["cardinal"].Value;
+            var monthStr = match.Groups["month"].Value;
+            var noYear = false;
+            int year;
+
+            int cardinal;
+            if (cardinalStr.Equals("最后一", StringComparison.Ordinal))
```

Comment:
Would a followup PR move these to YAML and update the relevant Chinese config and config interface?

Reply:
yes, in this PR we focused only on the refactoring of the Base CJK Extractor/Parser without touching the various methods used by each entity. The idea is that, as we proceed in the Japanese/Korean development, we will inspects these methods more closely and move any Chinese specific reference to the config files.